Wikipedia/Code/AppearanceSettingsViewController.swift

```diff
@@ -0,0 +1,245 @@
+        tableView.cellForRow(at: indexPath)?.accessoryType = .checkmark
+        item.checkmarkAction()
+    }
+    
+    public func tableView(_ tableView: UITableView, didDeselectRowAt indexPath: IndexPath) {
+        tableView.cellForRow(at: indexPath)?.accessoryType = .none
+    }
+    
+    public func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
+        let currentAppTheme = UserDefaults.wmf_userDefaults().wmf_appTheme
+        
+        if let checkmarkItem = sections[indexPath.section].items[indexPath.item] as? AppearanceSettingsCheckmarkItem {
+            
+            switch currentAppTheme {
+            case Theme.darkDimmed where checkmarkItem.title == "Dark":
+                fallthrough
+            case checkmarkItem.theme:
+                cell.accessoryType = .checkmark
+                tableView.selectRow(at: indexPath, animated: false, scrollPosition: .none)
+            default:
+                break
+            }
+    
+        }
+    }
+    
+    public func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        guard let item = sections[indexPath.section].items[indexPath.item] as? AppearanceSettingsCustomViewItem else {
+            return tableView.rowHeight
+        }
+        return item.viewController.view.frame.height
+    }
+    
+    func handleImageDimmingSwitchValueChange(_ sender: UISwitch) {
+        let currentTheme = UserDefaults.wmf_userDefaults().wmf_appTheme
+        UserDefaults.wmf_userDefaults().wmf_isImageDimmingEnabled = sender.isOn
+        userDidSelect(theme: currentTheme.withDimmingEnabled(sender.isOn))
+    }
+    
+    public func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        let header = WMFTableHeaderLabelView.wmf_viewFromClassNib()
+        if let th = header as Themeable? {
+            th.apply(theme: theme)
+        }
+        header?.text = sections[section].headerTitle
+        return header
+    }
+    
+    public func tableView(_ tableView: UITableView, viewForFooterInSection section: Int) -> UIView? {
+        let footer = WMFTableHeaderLabelView.wmf_viewFromClassNib()
+        if let th = footer as Themeable? {
+            th.apply(theme: theme)
+        }
+        footer?.setShortTextAsProse(sections[section].footerText)
+        return footer
+    }
+    
+    public func tableView(_ tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
+        guard let footerText = sections[section].footerText else {
+            return 0
+        }
+        
+        let footer = WMFTableHeaderLabelView.wmf_viewFromClassNib()
+        footer?.text = footerText
+        return footer!.height(withExpectedWidth: self.view.frame.width)
+    }
+    
+    public func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        let header = WMFTableHeaderLabelView.wmf_viewFromClassNib()
```

Comment:
Nevermind - the custom headers & footers can be replaced with the standard headers and footers. This also fixes the layout attribute issue. The custom headers only exist in Notification settings due to having non-uppercase text in a header.

So all the current viewForHeader/Footer and heightForHeader/Footer can be replaced with:

```
    public func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
        return sections[section].headerTitle
    }
    
    public func tableView(_ tableView: UITableView, titleForFooterInSection section: Int) -> String? {
        return sections[section].footerText
    }
```